pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/dependency_test.go

```diff
@@ -98,8 +98,23 @@ var _ = Describe("Resource Dependency in an ApplicationConfiguration", func() {
 	AfterEach(func() {
 		logf.Log.Info("Clean up resources")
 		// delete the namespace with all its resources
-		Expect(k8sClient.Delete(ctx, in)).Should(SatisfyAny(BeNil(), &util.NotFoundMatcher{}))
-		Expect(k8sClient.Delete(ctx, out)).Should(BeNil())
+		ac := &v1alpha2.ApplicationConfiguration{}
+		Expect(k8sClient.DeleteAllOf(ctx, ac, client.InNamespace(namespace))).Should(Succeed())
+		cm := &corev1.ConfigMap{}
+		Expect(k8sClient.DeleteAllOf(ctx, cm, client.InNamespace(namespace))).Should(Succeed())
+		foo := &unstructured.Unstructured{}
+		foo.SetAPIVersion("example.com/v1")
+		foo.SetKind("Foo")
+		Expect(k8sClient.DeleteAllOf(ctx, foo, client.InNamespace(namespace))).Should(Succeed())
+		Eventually(func() bool {
+			l := &unstructured.UnstructuredList{}
+			l.SetAPIVersion("example.com/v1")
+			l.SetKind("Foo")
+			if err := k8sClient.List(ctx, l, client.InNamespace(namespace)); err != nil {
+				return false
+			}
+			return len(l.Items) == 0
+		}, 3*time.Second, time.Second).Should(BeTrue())
```

Comment:
Why make this change? Move "cleaning up resource" operation into AfterEach to ensure the resources created by each ginkgo.It spec can be deleted even the spec failed. Otherwise, residual resources may block other cases.